mp/src/game/client/momentum/mom_system_discord.cpp

```diff
@@ -0,0 +1,512 @@
+        default:
+            return;
+    }
+}
+
+// Causes the current player to start spectating the target user
+void CMomentumDiscord::SpecPlayerFromSteamId(const char* steamID) {
+    DevWarning("CMomentumDiscord::SpecPlayerFromSteamId(%s)\n", steamID);
+
+    // MOM_TODO: We probably should check if the target player is already spectating someone, then set the spectate target them instead
+    // Or we could adjust the `mom_spectate` command to handle this automatically
+    DISPATCH_CON_COMMAND("mom_spectate", CFmtStrN<DISCORD_MAX_BUFFER_SIZE>("mom_spectate %s", steamID));
+}
+
+// Updates the current and max party size based upon the current lobby
+void CMomentumDiscord::UpdateLobbyNumbers() {
+    CHECK_STEAM_API(SteamMatchmaking());
+    if (m_sSteamLobbyID.IsValid()) {
+        m_iDiscordPartySize = SteamMatchmaking()->GetNumLobbyMembers(m_sSteamLobbyID);
+        m_iDiscordPartyMax = SteamMatchmaking()->GetLobbyMemberLimit(m_sSteamLobbyID);
+    }
+}
+
+// Keeps the discord party ID and secrets up to date from the steam data
+void CMomentumDiscord::UpdateDiscordPartyIdFromSteam() {
+    if (m_sSteamLobbyID.IsValid()) {
+        // The party ID is just the Steam Lobby ID
+        V_strncpy(m_szDiscordPartyId, CFmtStrN<DISCORD_MAX_BUFFER_SIZE>("%lld", m_sSteamLobbyID.ConvertToUint64()), DISCORD_MAX_BUFFER_SIZE);
+
+
+        if (!m_sSteamUserID.IsValid()) {
+            // If the steam user ID is not valid, try and get it
+            GetSteamUserID();
+        }
+
+        // We could do something to further "encrypt" the secrets but it's probably fine
+        if (m_sSteamUserID.IsValid()) {
+            // If we found a vaid steam user ID add it into the secret
+            V_strncpy(m_szDiscordJoinSecret, CFmtStrN<DISCORD_MAX_BUFFER_SIZE>("J%lld;%lld",
+                m_sSteamLobbyID.ConvertToUint64(),
+                m_sSteamUserID.ConvertToUint64()),
+            DISCORD_MAX_BUFFER_SIZE);
+        }
+        else {
+            // Otherwise just fall back to the steam lobby ID
+            V_strncpy(m_szDiscordJoinSecret, CFmtStrN<DISCORD_MAX_BUFFER_SIZE>("J%lld",
+                m_sSteamLobbyID.ConvertToUint64()),
+            DISCORD_MAX_BUFFER_SIZE);
+        }
+
+        if (InMap()) {
+            // If the user is in a map then add a valid spectate secret (the same as the join secret only with an 'S')
+            // MOM_TODO: Check the `mom_lobby_type` cvar first
+            m_szDiscordSpectateSecret[0] = 'S';
+            V_strncpy(&m_szDiscordSpectateSecret[1], &m_szDiscordJoinSecret[1], DISCORD_MAX_BUFFER_SIZE-1);
+        }
+        else {
+            m_szDiscordSpectateSecret[0] = '\0';
+        }
+    }
+    else {
+        m_sSteamLobbyID.Clear();
+        m_szDiscordPartyId[0] = '\0';
+        m_szDiscordJoinSecret[0] = '\0';
+    }
+}
+
+// Clears out all of the discord fields (optionally can preserve the party ID fields)
+void CMomentumDiscord::ClearDiscordFields(bool clearPartyFields /*=false*/) {
```

Comment:
Minor, it's actually defaulted to true here, but perhaps it should be a separate `Clear` method?